sys/od/od.c

```diff
@@ -27,12 +27,18 @@ void od_hex_dump(const void *data, size_t data_len, uint8_t width)
         width = OD_WIDTH_DEFAULT;
     }
 
+    uint8_t str_pos = 0;
     for (unsigned int i = 0; i < data_len; i++) {
-        print_str("  ");
+        print_str(" ");
         print_byte_hex(((uint8_t *)data)[i]);
 
         if ((((i + 1) % width) == 0) || i == (data_len - 1)) {
-            puts("");
+            for (unsigned j = i; ((j + 1) % width) != 0; ++j) {
+                print_str("   ");
+            }
+            printf("  %.*s\n", (i % width) + 1, (char *)data + str_pos);
```

Comment:
The output of this will be broken, if `data` contains `\0` byte (because it isn't a pure complete string e.g.)

Reply:
>if data contains \0 byte (because it isn't a pure complete string e.g.)

Hm that's true. Printing the characters separately would make mor sense here. And, I think @miri64 proposed in another comment, a check should be done whether the character is actually printable and use e.g. a `.` if not.

Reply:
For reference it is in https://github.com/RIOT-OS/RIOT/pull/7516#issuecomment-334670202